nmostesting/suites/IS0703Test.py

```diff
@@ -0,0 +1,119 @@
+                self.transport_types[sender] = "urn:x-nmos:transport:rtp"
+
+        if len(self.is05_senders) > 0:
+            for sender in self.is05_senders:
+                dest = "single/senders/" + sender + "/active"
+                valid, response = self.is05_utils.checkCleanRequestJSON("GET", dest)
+                if valid:
+                    if len(response) > 0 and isinstance(response["transport_params"][0], dict):
+                        self.sender_active_params[sender] = response["transport_params"][0]
+
+    def test_01(self, test):
+        """Each Sender has the required ext parameters"""
+
+        ext_params_websocket = ['ext_is_07_source_id', 'ext_is_07_rest_api_url']
+        ext_params_mqtt = ['ext_is_07_rest_api_url']
+        if len(self.senders_to_test.keys()) > 0:
+            for sender in self.senders_to_test:
+                if sender in self.sender_active_params:
+                    all_params = self.sender_active_params[sender].keys()
+                    params = [param for param in all_params if param.startswith("ext_")]
+                    valid_params = False
+                    if self.transport_types[sender] == "urn:x-nmos:transport:websocket":
+                        if sorted(params) == sorted(ext_params_websocket):
+                            valid_params = True
+                    elif self.transport_types[sender] == "urn:x-nmos:transport:mqtt":
+                        if sorted(params) == sorted(ext_params_mqtt):
+                            valid_params = True
+                    if not valid_params:
+                        return test.FAIL("Missing required ext parameters for Sender {}".format(sender))
+            return test.PASS()
+        else:
+            return test.UNCLEAR("Not tested. No resources found.")
+
+    def test_02(self, test):
+        """Each Source has a corresponding IS-05 sender"""
+
+        if len(self.is07_sources) > 0:
+            if len(self.sources_to_test.keys()) > 0:
+                for source in self.is07_sources:
+                    if source in self.sources_to_test:
+                        found_source = False
+                        for sender in self.senders_to_test:
+                            if sender in self.sender_active_params:
+                                try:
+                                    if self.sender_active_params[sender]["ext_is_07_source_id"] == source:
+                                        found_source = True
+                                except KeyError as e:
+                                    return test.FAIL("Sender {} parameters do not contain expected key: {}"
+                                                     .format(sender, e))
+                        if not found_source:
+                            return test.FAIL("Source {} has no associated IS-05 sender".format(source))
+                return test.PASS()
+            else:
+                return test.UNCLEAR("Not tested. No resources found.")
```

Comment:
Instead of UNCLEAR?

Reply:
Yes, since we have IS-07 Sources but no IS-04 Sources to test. Perhaps with a message like `"Source {} not found in Node API".format(self.is07_sources[0])`? (Not quite sure of the syntax there.)

Reply:
Ok. Understood